cmake/CMakeLists.txt

```diff
@@ -76,19 +78,71 @@ set(PUBLIC_HEADERS
   ${ROOT_DIR}/include/winsparkle.h
   ${ROOT_DIR}/include/winsparkle-version.h)
 
-set(RESOURCES winsparkle translations)
-
-# resource compilation for MinGW
-foreach(RESOURCE ${RESOURCES})
-  add_custom_command(OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/${RESOURCE}.o
-                     COMMAND ${CMAKE_RC_COMPILER} -I${CMAKE_CURRENT_SOURCE_DIR} -i${CMAKE_CURRENT_SOURCE_DIR}/${RESOURCE}.rc
-                             -o ${CMAKE_CURRENT_BINARY_DIR}/${RESOURCE}.o)
+set(LANGUAGES
```

Comment:
This is yet another place where languages list has to be maintained (and manually). Is there no way in CMake to look for all available *.po files instead?

Reply:
You could use `file(GLOB VARIABLE ${ROOT_DIR}/translations/*.po CONFIGURE_DEPENDS)` (https://cmake.org/cmake/help/v3.12/command/file.html#glob), but it was new in CMake 3.12 so you'd have to rause your cmake_minimum_required quite a bit. Before that there was no way to get CMake to re-configure if the glob results change. I'm not sure if 2.8 (which is your current minimum) even has file(GLOB) at all.

Reply:
@puetzk Thanks! How problematic is raising CMake version? I think not much on Windows where distro packaging doesn't enter the picture… AppVeyor, which I care about for CI, already includes 3.13.

Reply:
@puetzk However not using CMake myself, I don't really know how to translate your advice into actual code; a patch/PR would be tremendously helpful.

Reply:
[file(GLOB)](https://cmake.org/cmake/help/v2.8.12/cmake.html#command:file) available at least since CMake 2.8, so no raising is required.

Reply:
But that's missing CONFIGURE_DEPENDS (which is what was new in 3.12), so it won't reconfigure properly if the filesystem changes.

Reply:
So if and only if:
- already run cmake command (configure build/generate build scripts);
- after that you add some new _*.po_ file(s);
- and run cmake --build (or make, if this is mingw build, or build in Visual Studio);
In that case the build won't include new files.

If you run cmake (configure) command manually (which you normally do, if you have CI, or smth) - everything works.
If you remove existing .po file - build reasonably fails, until you run cmake.
If you do a clean build - everything works by default anyway.

Don't think this small feature is worth considering raising minimal required cmake version, specially for build system that is not officially supported.

I've seen two approaches, which I'd support widely:
1) Support cmake 2.8 and higher - great if you want to have(already have) a wide range of supported platforms/clients;
2) Support only latest cmake release - great for development, specially if you control all the build environments;

Anything in between sounds like not a great compromise.